swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt

```diff
@@ -552,6 +553,54 @@ abstract class AbstractResourceTypeHandler<T : Resource>(
     }
   }
 
+  /**
+   * Evaluates a resource against rules and returns violation summaries for this resource
+   * @see [com.netflix.spinnaker.config.ResourceTypeConfiguration.RuleConfiguration]
+   * @see [WorkConfiguration.enabledRules]
+   */
+  private fun T.getViolations(workConfiguration: WorkConfiguration): List<Summary> {
+    if (workConfiguration.enabledRules.isNullOrEmpty()) {
+      // Evaluate the resource against all rules and get violations
+      return getViolations()
```

Comment:
do we want to evaluate all the all rules in this case vs not run any of them , trying to understand the reason here ?

Reply:
If there are no enabled rules it will fallback to the usual behavior. I added unit tests to illustrate that.